File: tests/test_pages_build.py
import shutil
import subprocess
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
OUTPUT = ROOT / "dist"


def test_pages_build_contains_advanced_worker_and_runtime_module():
    shutil.rmtree(OUTPUT, ignore_errors=True)
    subprocess.run(["npm", "run", "build:pages"], cwd=ROOT, check=True)

    assert (OUTPUT / "_worker.js").is_file()
    assert (OUTPUT / "default-pages.mjs").is_file()
    assert (OUTPUT / "admin.js").is_file()
    assert (OUTPUT / "styles.css").is_file()
    assert (OUTPUT / "push-sw.js").is_file()
    assert (OUTPUT / "manifest.webmanifest").is_file()
    assert (OUTPUT / "web-push-browser" / "index.js").is_file()

    worker = (OUTPUT / "_worker.js").read_text()
    assert "from './default-pages.mjs'" in worker
    assert "from './web-push-browser/index.js'" in worker
    assert "dashboard-welcome" in worker
    assert "data-notify-me" in worker
    assert "Trevor Olsen" not in worker
    # Legacy CMS pages-list tab should stay gone (do not flag zernio-facebook-pages-list).
    assert 'id="pages-list"' not in worker
    assert "#pages-list" not in worker

File: tests/test_pages_build.mjs
import assert from 'node:assert/strict';
import { spawnSync } from 'node:child_process';
import { existsSync, readFileSync, rmSync } from 'node:fs';
import { join } from 'node:path';
import { test } from 'node:test';
import { fileURLToPath } from 'node:url';

const ROOT = join(fileURLToPath(new URL('.', import.meta.url)), '..');
const OUTPUT = join(ROOT, 'dist');

test('pages build contains advanced worker and runtime module', () => {
  rmSync(OUTPUT, { recursive: true, force: true });
  const sync = spawnSync('node', ['worker/scripts/sync-public.mjs'], { cwd: ROOT, encoding: 'utf8' });
  assert.equal(sync.status, 0, sync.stderr || sync.stdout);
  const build = spawnSync('node', ['worker/scripts/build-pages.mjs'], { cwd: ROOT, encoding: 'utf8' });
  assert.equal(build.status, 0, build.stderr || build.stdout);

  assert.equal(existsSync(join(OUTPUT, '_worker.js')), true);
  assert.equal(existsSync(join(OUTPUT, 'default-pages.mjs')), true);
  assert.equal(existsSync(join(OUTPUT, 'admin-audit-log.mjs')), true);
  assert.equal(existsSync(join(OUTPUT, 'inkind-forms.mjs')), true);
  assert.equal(existsSync(join(OUTPUT, 'admin.js')), true);
  assert.equal(existsSync(join(OUTPUT, 'styles.css')), true);
  assert.equal(existsSync(join(OUTPUT, 'push-sw.js')), true);
  assert.equal(existsSync(join(OUTPUT, 'manifest.webmanifest')), true);
  assert.equal(existsSync(join(OUTPUT, 'web-push-browser/index.js')), true);
  assert.equal(existsSync(join(OUTPUT, 'vendor/jspdf.umd.min.js')), true);

  const worker = readFileSync(join(OUTPUT, '_worker.js'), 'utf8');
  assert.match(worker, /from '\.\/default-pages\.mjs'/);
  assert.match(worker, /from '\.\/admin-audit-log\.mjs'/);
  assert.match(worker, /from '\.\/inkind-forms\.mjs'/);
  assert.match(worker, /from '\.\/web-push-browser\/index\.js'/);
  assert.match(worker, /dashboard-welcome/);
  assert.match(worker, /data-notify-me/);
  assert.match(worker, /web_push_subscriptions/);
  assert.doesNotMatch(worker, /Trevor Olsen/);
  // Legacy CMS pages-list tab should stay gone (do not flag zernio-facebook-pages-list).
  assert.doesNotMatch(worker, /id=["']pages-list["']/);
  assert.doesNotMatch(worker, /#pages-list\b/);
});
